smtp.go

```diff
@@ -0,0 +1,94 @@
+package emaildriver
+
+import (
+	"bytes"
+	"errors"
+	"net"
+	"strconv"
+
+	"gopkg.in/gomail.v2"
+)
+
+type SMTPEmailBackend struct {
+	isEnabled bool
+	config
+}
+
+type config struct {
+	host string
+	port int
+	user string
+	pass string
+}
+
+// Init receives the configs, validates them and sets on the SMTPEmailBackend struct for use by Send function
+func (m *SMTPEmailBackend) Init(conf map[string]string) error {
+
+	if conf == nil {
+		return errors.New("SmtpEmailBackend requires a config map")
+	}
+
+	user, ok := conf["SMTPUsername"]
+	if !ok || user == "" {
+		return errors.New("SMTPUsername not defined")
+	}
+
+	pass, ok := conf["SMTPPassword"]
+	if !ok || pass == "" {
+		return errors.New("SMTPPassword not defined")
+	}
+
+	host, port, err := net.SplitHostPort(conf["SMTPAddress"])
+	if err != nil {
+		return err
+	}
+
+	m.host = host
+	m.port, _ = strconv.Atoi(port)
+	m.user = user
+	m.pass = pass
+	m.isEnabled = true
+
+	return nil
+}
+
+func (m *SMTPEmailBackend) Send(emailMeta EmailMeta, emailData interface{}, textTemplateName TykTemplateName,
+	htmlTemplateName TykTemplateName, OrgId string, Styles string) error {
+
+	if !m.isEnabled {
+		log.Warning("SMTP driver not initialized.")
+		return errors.New("driver not initialised correctly")
+	}
+
+	// Generate strings from template
+	var txtDoc bytes.Buffer
+
+	type superEmailData struct {
+		Data   interface{}
+		Styles string
+	}
+
+	thisData := superEmailData{Data: emailData}
+	thisData.Styles = Styles
+
+	if err := PortalEmailTemplatesTXT.ExecuteTemplate(&txtDoc, string(textTemplateName), emailData); err != nil {
+		log.Error("error executing text template:", err)
+		return err
+	}
+
+	msg := gomail.NewMessage()
+	msg.SetHeader("From", emailMeta.FromEmail, emailMeta.FromName)
+	msg.SetHeader("To", emailMeta.RecipientEmail)
+	msg.SetHeader("Subject", emailMeta.Subject)
+	msg.SetBody("text/plain", txtDoc.String())
+
+	dialer := gomail.NewDialer(m.host, m.port, m.user, m.pass)
+
+	if err := dialer.DialAndSend(msg); err != nil {
+		log.Error("error sending mail:", err)
+		return err
+	}
+
+	log.Debugf("email sent addr: %s, from: %s, to: %s", m.host, emailMeta.FromEmail, emailMeta.RecipientEmail)
```

Comment:
might be better to use `log.WithFields(...).Debug("email sent")` as logging library is supposed to simplify format output code like this